internal/dot/graph.go

```diff
@@ -252,6 +318,87 @@ func (dg *Graph) addToGroup(r *Result, id CtorID) {
 	group.Results = append(group.Results, r)
 }
 
+// Prune removes elements from the graph that do not have failed results.
+// Removing elements that do not have failing results makes the graph easier to debug, since non-failing
+// nodes and edges can clutter the graph and don't help the user debug.
+func (dg *Graph) PruneSuccess() {
+	dg.pruneCtors()
+	dg.pruneGroups()
+}
+
+// Remove constructors from the graph that do not have failing Results.
+func (dg *Graph) pruneCtors() {
+	var pruned []*Ctor
+	for _, c := range dg.Ctors {
+		if _, ok := dg.Failed.ctors[c.ID]; ok {
+			pruned = append(pruned, c)
+		} else {
+			dg.pruneCtorParams(c)
+			dg.pruneGroupResults(c)
+			delete(dg.ctorMap, c.ID)
+		}
+	}
+
+	dg.Ctors = pruned
+}
+
+// pruneGroups removes groups from the graph that do not have failing results.
+func (dg *Graph) pruneGroups() {
+	var pruned []*Group
+	for _, g := range dg.Groups {
+		k := g.groupKey()
+		if _, ok := dg.Failed.groups[k]; ok {
+			pruned = append(pruned, g)
+		} else {
```

Comment:
```suggestion
			continue
		}

```

Reply:
I don't think I can just add continue since I still need to delete the entry from the groupMap

Reply:
Offline discussion: I understand the comment, will change.